### PR TITLE
Update to Play 2.6

### PR DESCRIPTION
**Pull request by marcospereira**
Also:

1. Update sbt and scala versions
1. Add more comprehensive examples and tests to sample app
1. Remove uses os deprecated APIs

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -8,7 +8,8 @@ lazy val plugin = Project (
     name := "play-scalate",
     organization := "com.github.tototoshi",
     version := "0.3.0",
-    scalaVersion := "2.11.8",
+    scalaVersion := "2.12.4",
+    crossScalaVersions := Seq("2.11.12", "2.12.4"),
```
Comment (marcospereira) [inline build.sbt]: Play 2.6 cross-builds to Scala 2.11 and 2.12. So this module needs to be published to both versions.

File: build.sbt [hunk 2]
```diff
@@ -27,19 +28,21 @@ lazy val playapp = Project(
...
   resourceDirectories in Test += baseDirectory.value / "conf",
-  scalaVersion := "2.11.8",
+  scalaVersion := "2.12.4",
   version := playAppVersion,
   libraryDependencies ++= Seq(
+    guice,
     "org.scalatra.scalate" %% "scalate-core" % "1.8.0",
-    "org.scala-lang" % "scala-compiler" % scalaVersion.value
+    "org.scala-lang" % "scala-compiler" % scalaVersion.value,
+    "org.scalatestplus.play" %% "scalatestplus-play" % "3.1.2" % Test
```
Comment (marcospereira) [inline build.sbt:38]: scalatestplus-play is used to test playapp controllers.

**Comment by marcospereira** [comment]
Just notice this overlaps with #4 and #5.

You may want first to merge them, and I will rebase and solve the conflicts for this one.

**Comment by seratch** [comment]
@marcospereira Thanks. I've merged them. Could you rebase this branch? Sorry for bothering you.

**Comment by marcospereira** [comment]
@seratch rebased!

**Pull request merged**